data_diff/diff_tables.py

```diff
@@ -31,33 +32,50 @@ def split_space(start, end, count):
     return list(range(start, end, (size + 1) // (count + 1)))[1 : count + 1]
 
 
+def parse_table_name(t):
+    return tuple(t.split("."))
+
+
 @dataclass(frozen=False)
 class TableSegment:
+    """Signifies a segment of rows (and selected columns) within a table"""
+
+    # Location of table
     database: Database
     table_path: DbPath
+
+    # Name of the key column, which uniquely identifies each row (usually id)
     key_column: str
+
+    # Name of updated column, which signals that rows changed (usually updated_at or last_update)
     update_column: str = None
+
+    # Extra columns to compare
     extra_columns: Tuple[str, ...] = ()
-    start_key: DbKey = None
-    end_key: DbKey = None
-    min_time: DbTime = None
-    max_time: DbTime = None
+
+    # Start/end key_column values, used to restrict the segment
+    min_key: DbKey = None
+    max_key: DbKey = None
+
+    # Start/end update_column values, used to restrict the segment
+    min_update: DbTime = None
+    max_update: DbTime = None
 
     def __post_init__(self):
-        if not self.update_column and (self.min_time or self.max_time):
-            raise ValueError("Error: min_time/max_time feature requires to specify 'update_column'")
+        if not self.update_column and (self.min_update or self.max_update):
+            raise ValueError("Error: min_update/max_update feature requires to specify 'update_column'")
 
     def _make_key_range(self):
-        if self.start_key is not None:
-            yield Compare("<=", str(self.start_key), self.key_column)
-        if self.end_key is not None:
-            yield Compare("<", self.key_column, str(self.end_key))
+        if self.min_key is not None:
+            yield Compare("<=", str(self.min_key), self.key_column)
+        if self.max_key is not None:
+            yield Compare("<", self.key_column, str(self.max_key))
 
     def _make_update_range(self):
```

Comment:
The last confusion caused by our relative parsing could likely be fixed by erroring with a helpful error message if `min_update > max_update`